src/middleware/log_request.rs

```diff
@@ -31,19 +31,37 @@ impl Handler for LogRequests {
         let response_time =
             response_time.as_secs() * 1000 + u64::from(response_time.subsec_nanos()) / 1_000_000;
 
-        print!(
-            "at={level} method={method} path=\"{path}\" \
-             request_id={request_id} fwd=\"{ip}\" service={time_ms}ms \
-             status={status} user_agent=\"{user_agent}\"",
-            level = level,
-            method = req.method(),
-            path = FullPath(req),
-            ip = request_header(req, "X-Real-Ip"),
-            time_ms = response_time,
-            user_agent = request_header(req, "User-Agent"),
-            request_id = request_header(req, "X-Request-Id"),
-            status = response_code,
-        );
+        if let Some(metadata_length) = req.extensions().find::<u64>() {
```

Comment:
This adds a lot of duplication. What do you think about, before the `print`, doing something like:

```
let metadata_length = req.extensions().find::<u64>().map_or(String::new(), |n| format!(" metadata_length={}", n));
```

which will set `metadata_length` to an empty string if there isn't a length, and will set `metadata_length` to ` metadata_length=32` if the length is `Some(32)`.

Then the `print` can unconditionally print out the value in `metadata_length`.

Does that make sense? If not, please ask questions!

Reply:
this makes perfect sense, I was wondering myself how to reduce the duplicity of the branches, and I am not very familiar with map/reduce type things, so this will be an excellent change